Evolved Orb Items

3xtazzy opened the pull request:
AggroKill method

3xtazzy committed:
Update CoreHollowbornChaosEnvoy.cs

AggroKill method

3xtazzy committed:
Merge branch 'Skua' of https://github.com/BrenoHenrike/Scripts into Skua

3xtazzy committed:
QuestRewards method, and use of getAll for recursive

3xtazzy committed:
Includes

3xtazzy committed:
Merge branch 'BrenoHenrike:Skua' into Skua

3xtazzy committed:
Merge branch 'Skua' of https://github.com/BrenoHenrike/Scripts into Skua

3xtazzy committed:
Merge branch 'Skua' of https://github.com/3xtazzy/Scripts into Skua

3xtazzy committed:
Missed Aggrokill

3xtazzy committed:
EbilHQ: Story added

3xtazzy committed:
HBChaos: Items > bank after quest done

3xtazzy committed:
Frostvale AC Gifting quest

LordExelot commented on Hollowborn/HollowbornChaosEnvoy/CoreHollowbornChaosEnvoy.cs:
```diff
@@ -253,6 +264,33 @@ public void PersistingMayhem(bool getAll = true)
         Core.CancelRegisteredQuests();
     }
+    
+    /// <summary>
+    /// Joins a map and aggro, then kill the monsters
+    /// Do not aggro too many monsters
+    /// </summary>
+    /// <param name="map">Map to join</param>
+    /// <param name="monsterNames">Name of the monsters to kill. Recommended to pick monster with highest quantity and lowest stats</param>
+    /// <param name="cell">Cell to jump to. Recommended to pick cell with most monsters</param>
+    /// <param name="item">Item to hunt the monster for, if null will just hunt & kill the monster 1 time</param>
+    /// <param name="quant">Desired quantity of the item</param>
+    /// <param name="isTemp">Whether the item is temporary</param>
+    /// <param name="pad">Pad to jump to. Set default to Spawn because needed to bypass the Army restriction</param>
+    public void AggroKill(string map, string [] monsterNames, string cell, string item, int quant, bool isTemp = true, string pad = "Spawn")
+    {
+        if (item != null && (isTemp ? Bot.TempInv.Contains(item, quant) : Core.CheckInventory(item, quant)))
+            return;
+        if (!isTemp && item != null)
+            Core.AddDrop(item);
+        
+        Core.Join(map, cell, pad);
+        Army.AggroMonNames(monsterNames);
+        Army.AggroMonStart(map);
+        while (!Bot.ShouldExit && (isTemp ? !Bot.TempInv.Contains(item, quant) : !Core.CheckInventory(item, quant)))
+            Bot.Combat.Attack("*");
+        Army.AggroMonStop(true);
+    }
```
Denied, as a standalone bot it's not intended to be using aggro methods, people without army would get in trouble for this. Most of our user base doesn't use army

LordExelot commented on Seasonal/HarvestDay/MergeShops/EbilCorpHQMerge.cs:
```diff
@@ -25,6 +26,7 @@ public void ScriptMain(IScriptInterface Bot)
         Core.BankingBlackList.AddRange(NeededItems);
         Core.SetOptions();
 
+        HarvestDay.EbilCorpHQ();
```
Please move this inside the merge function, script main must remain empty as per format

LordExelot commented on Seasonal/HarvestDay/MergeShops/EbilCorpHQMerge.cs:
```diff
@@ -91,7 +93,7 @@ void findIngredients()
                     {
-                        Core.HuntMonster("ebilcorphq", "Master Chairman", req.Name, isTemp: false, log:false);
+                        Core.HuntMonster("ebilcorphq", "Master Chairman", req.Name, isTemp: false, log: false);
                         Core.Logger("This item is not setup yet");
```
Why is this log here

LordExelot reviewed (commented):
Don't go making non army bots into army bots mate

3xtazzy committed:
Revert HBChaos, logging in EbilHQ

3xtazzy committed:
Removed army includes

3xtazzy commented:
> Denied, as a standalone bot it's not intended to be using aggro methods, people without army would get in trouble for this. Most of our user base doesn't use army 

Logging added

3xtazzy commented:
> Denied, as a standalone bot it's not intended to be using aggro methods, people without army would get in trouble for this. Most of our user base doesn't use army 

Removed and reverted back

3xtazzy commented:
> Please move this inside the merge function, script main must remain empty as per format  

Moved to the function

3xtazzy committed:
EvolvedShadowOrb: Items
